testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractFirstBrokerLoginTest.java

```diff
@@ -286,14 +288,19 @@ public void testLinkAccountByLogInAsUserUsingBrowserButtons() {
         // Click browser 'back' and then 'forward' and then continue
         driver.navigate().back();
         assertTrue(driver.getPageSource().contains("You are already logged in."));
-        driver.navigate().forward();
+        driver.navigate().forward(); // here a new execution ID is added to the URL using JS, see below
         idpConfirmLinkPage.assertCurrent();
 
         // Click browser 'back' on review profile page
         idpConfirmLinkPage.clickReviewProfile();
         waitForPage(driver, "update account information", false);
         updateAccountInformationPage.assertCurrent();
         driver.navigate().back();
+        // JS-capable browsers (i.e. all except HtmlUnit) add a new execution ID to the URL which then causes the login expire page to appear (because the old ID and new ID don't match)
+        if (!(driver instanceof HtmlUnitDriver)) {
+            loginExpiredPage.assertCurrent();
+            loginExpiredPage.clickLoginContinueLink();
+        }
```

Comment:
I'm not sure if this is intended behavior. @mposolda Could you please double check this?

Reply:
@vmuzikar Yes, this is expected behaviour. IMO this change is fine

Reply:
Thanks!